pkg/testing/integration/program.go

```diff
@@ -2008,8 +2008,16 @@ func (pt *ProgramTester) prepareDotNetProject(projinfo *engine.Projinfo) error {
 
 	localNuget := os.Getenv("PULUMI_LOCAL_NUGET")
 	if localNuget == "" {
-		home := os.Getenv("HOME")
-		localNuget = filepath.Join(home, ".pulumi", "nuget")
+		// We assume that the test is run from a subdirectory of Pulumi. We then search backwards
+		// until we are in a directory called Pulumi.
```

Comment:
I'm not sure we can make this assumption. The `integration` package is used by other repos, and potentially repos not in the github.com/pulumi org. I guess in those cases `PULUMI_LOCAL_NUGET` would have to be set?

Reply:
I agree that the assumption is unsafe. We also have the scenario where someone hacks on Pulumi by putting it into a `pulumi-dev` (or `Pulumi`) folder. The tests will fail there as well. The problem is that we want to put `PULUMI_ROOT` (and thus `PULUMI_LOCAL_NUGET`) in a relative directory. 

Off of the top of my head, I can't think of a way to avoid something like this and still use a relative path. As an alternative, we could use `$HOME/.pulumi/.dev` as `PULUMI_ROOT`. This would avoid this questionable lookup and also allow maintaining both a development and released instance of pulumi.